it/src/main/java/org/corfudb/universe/node/server/vm/VmCorfuServer.java

```diff
@@ -0,0 +1,256 @@
+                universeParams.getVmPassword(),
+                CORFU_INFRASTRUCTURE_JAR,
+                "./" + params.getName() + "/corfu-server.jar"
+        );
+
+        start();
+
+        return this;
+    }
+
+    /**
+     * Disconnect the VM from the vSphere network
+     */
+    @Override
+    public void disconnect() {
+        log.info("Disconnecting the server from the network. VM: {}", params.getVmName());
+
+        universeParams.getVmIpAddresses().values().stream()
+                .filter(addr -> !addr.equals(getIpAddress()))
+                .forEach(addr -> {
+                    executeSudoCommand(IpTablesUtil.dropInput(addr));
+                    executeSudoCommand(IpTablesUtil.dropOutput(addr));
+                });
+    }
+
+    /**
+     * Pause the {@link CorfuServer} process on the VM
+     */
+    @Override
+    public void pause() {
+        log.info("Pausing the Corfu server: {}", params.getName());
+
+        String cmd = "ps -ef | grep -v grep | grep \"corfudb\" | awk '{print $2}' | xargs kill -STOP";
+        executeCommand(cmd);
+    }
+
+    /**
+     * Start a {@link CorfuServer} process on the VM
+     */
+    @Override
+    public void start() {
+        // Compose command line for starting Corfu
+        StringBuilder cmdLine = new StringBuilder();
+        cmdLine.append("sh -c 'nohup java -cp ./" + params.getName() + "/*.jar org.corfudb.infrastructure.CorfuServer ");
+        cmdLine.append(getCommandLineParams());
+        cmdLine.append(" > /tmp/corfu.log 2>&1 &'");
+
+        executeCommand(cmdLine.toString());
+    }
+
+    /**
+     * Restart the {@link CorfuServer} process on the VM
+     */
+    @Override
+    public void restart() {
+        stop(params.getStopTimeout());
+        start();
+    }
+
+    /**
+     * Reconnect a {@link Node} to the vSphere network
+     */
+    @Override
+    public void reconnect() {
+        log.info("Reconnecting the corfu server to the network. VM: {}", params.getVmName());
+
+        executeSudoCommand(String.join(" ", IpTablesUtil.cleanInput()));
+        executeSudoCommand(String.join(" ", IpTablesUtil.cleanOutput()));
+    }
+
+    /**
+     * Resume a {@link CorfuServer}
+     */
+    @Override
+    public void resume() {
+        log.info("Resuming the corfu server: {}", params.getName());
+
+        String cmd = "ps -ef | grep -v grep | grep \"corfudb\" | awk '{print $2}' | xargs kill -CONT";
+        executeCommand(cmd);
+    }
+
+    /**
+     * Executes a certain command on the VM.
+     */
+    private void executeCommand(String cmdLine) {
+        String ipAddress = getIpAddress();
+
+        commandHelper.executeCommand(ipAddress,
+                universeParams.getVmUserName(),
+                universeParams.getVmPassword(),
+                cmdLine
+        );
+    }
+
+    /**
+     * Executes a certain Sudo command on the VM.
+     */
+    private void executeSudoCommand(String cmdLine) {
+        String ipAddress = getIpAddress();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "ipAddress" which hides the field declared at line 35. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)